src/hipscat/io/paths.py

```diff
@@ -295,3 +295,19 @@ def get_partition_join_info_pointer(catalog_base_dir: FilePointer) -> FilePointe
         File Pointer to the catalog's `partition_join_info.csv` association metadata file
     """
     return append_paths_to_pointer(catalog_base_dir, PARTITION_JOIN_INFO_FILENAME)
+
+
+def is_valid_catalog(pointer: FilePointer) -> bool:
+    """Checks if there is a catalog_info and a partition_info file
+    for a given base catalog pointer
+
+    Args:
+        pointer: pointer to base catalog directory
+
+    Returns:
+        True if the catalog_info and partition_info files exist for
+        the catalog, False if not
+    """
+    catalog_info_exists = is_regular_file(get_catalog_info_pointer(pointer))
+    partition_info_exists = is_regular_file(get_partition_info_pointer(pointer))
```

Comment:
It'd be nice to load these files and make sure the format is correct. I think initializing a catalog info object will do that for you (It'll raise a ValueError if it's wrong). There's a function to load the correct catalog info object at `hipscat.catalog.dataset.catalog_info_factory.from_catalog_dir`.

Reply:
Thanks, that makes sense! I have initialized the catalog info object like you suggested and added a simple assertion on the test case that ensures an error is thrown if, for example, the catalog type is invalid.